### PR TITLE
Fixed incorrect client side rest method url generation when pathparam is inside url

### PR DESCRIPTION
**Pull request by asterixtintin**
…mitted

### CODE REVIEW

**Comment by konsoletyper** [comment]
I approve, this fix is correct. However, I have some requirements:

1. Can you please provide the test? See existing one in `RestIT`, you can add more tests there.
2. Please, rename this commit, I don't want to commit with description "attemp to" appear in master. This commit fixes the issue, so just claim it.

As for tests, I tried this one:

```java
    @Test
    public void passesPathParamInsideUrl() throws Exception {
        assertEquals(1, service.sum(10, 7, 3));
    }
```

It passes with your fix, but fails otherwise.

also, I had to fix `TestServiceImpl.java` correspondingly, i.e.:

```java
    @Override
    public int sum(int mod, int a, int b) {
        return (a + b) % mod;
    }
```

**Comment by asterixtintin** [comment]
Will do

**Comment by asterixtintin** [comment]
Hi,
I ran your suggestions today and for me, the tests pass with or without my changes.
Infact the RestIT tests pass no matter what. For example this passes: ` assertEquals(2, service.sum(10, 7, 3));` . Is there any process I need to follow to run tests properly?

**Comment by konsoletyper** [comment]
That's strange. I just tried and it fails with `org.teavm.flavour.rest.HttpStatusException: 404 Not Found`. It's correct, since due to bug fixed by your commit TeaVM performs request to the wrong URL. When I apply your fix, test still fails since `sum` is implemented improperly. Fixing both method implementation and assertion makes test pass.

Did you try to alter this assertion: `assertEquals(5, service.sum(2, 3))` to `assertEquals(6, service.sum(2, 3))`. Does test still pass? How do you run tests? I use `mvn clean verify`.

**Comment by asterixtintin** [comment]
I was able to run the tests as expected via console. The IDE was running incorrect command for the tests

File: rest/src/test/java/org/teavm/flavour/rest/test/RestIT.java [hunk 1]
```diff
@@ -26,13 +26,20 @@
 @RunWith(TeaVMTestRunner.class)
 @SkipJVM
 public class RestIT {
+
```
Comment (konsoletyper) [inline rest/src/test/java/org/teavm/flavour/rest/test/RestIT.java]: It's not something crical, but why did you add these empty lines? It's just an unnecessary change

Comment (asterixtintin) [reply rest/src/test/java/org/teavm/flavour/rest/test/RestIT.java]: Agreed. Was not intentional, the IDE applied my global Java formatting on save automatically
